cpp/src/arrow/flight/client.cc

```diff
@@ -993,6 +998,30 @@ class FlightClient::FlightClientImpl {
     return Status::OK();
   }
 
+  Status AuthenticateBasicToken(std::string username, std::string password,
```

Comment:
Should username/password be passed by const ref?

Reply:
yes